enhancements/windows-containers/container-runtime-containerd.md

```diff
@@ -0,0 +1,213 @@
+*Containerd has become the default runtime in linux/windows nodes.
+*Testing and adaptation will require less engineering effort compare to other solutions due to wider open source
+ community and enterprise support.
+
+### Design Details
+
+we plan to use containerd 1.6.0 to integrate into WMCO. Containerd will be installed as a first service before 
+kubelet installation as kubelet must require containerd to be running. We already bundled containerd as part of 
+WMCO package. As we can use ctr commandline to test containerd, crictl commandline will not be installed.
+Once containerd is installed, the rest of the service installation steps remain the same. Kubelet's dependency on Docker
+runtime will be removed and replaced by containerd. A separate folder will be created under C:\k\ to store containerd
+config and related files. In BYOH case, even after containerd becomes default runtime Docker will continue to run.
+User can choose to uninstall Docker or use it for non-kubernetes related requirements. As docker installation is
+handled by the user, it is up to the user to decide if Docker should continue to run or not.
+When it comes to MachineSet, we can choose a VM image which doesn't have Docker. Containerd will be the only runtime.
+Even if Docker gets installed at any time, it can co-exist with containerd. 
+steps to install containerd as service
+    *scp containerd/related executables into Windows VM
+    *copy the files in appropriate folder
+    *create containerd config file
+    *run below commands in PowerShell cmdlet
+    *run extra command for performance `Add-MpPreference -ExclusionProcess "$Env:containerd.exe"`
+    *register containerd as service
+    *start containerd as a Windows service
+
+## Network changes
+Current CNI/IPAM will be used for containerd and no change in HNS-Network and HNS-Endpoint creation
+steps. The config file which will be used by containerd will point to same CNI/IPAM executables.
+
+## WMCO feature flag
+Containerd feature flag will be used to enable this feature. This will be enabled by default in
+community operator for customers and developers to try it out.  This will become the default for OpenShift
+4.11. At any given point in time, we have one runtime support and do not allow switching between runtime.
+
+## logging
+Containerd can be started with parameters in which we can enable logging and specify the file path to
+log warnings/errors. Log files will be stored at c:\var\log\containerd
+
+## upgrade
+When a cluster is upgraded OLM will switch to using a new Red Hat operators
+index. Because WMCO is named the same in both indexes, OLM will upgrade WMCO
+from the previous version, up to the latest version available in the new
+cluster. For further details please refer [WMCO-Upgrade](https://github.com/openshift/enhancements/blob/master/enhancements/windows-containers/windows-machine-config-operator-upgrades.md)
```

Comment:
This explanation is not needed here. Just referencing the other enhancement is sufficient.